src/Symfony/Component/Cache/Adapter/FilesystemTagAwareAdapter.php

```diff
@@ -137,11 +129,6 @@ protected function doInvalidate(array $tagIds): bool
         return true;
```

Comment:
`doInvalidate` leaks inodes by not clearing invalidated tags, but this cannot be fixed with the current storage layout. We'd need a file listing item files instead of symlinks, as suggested in #33924. /cc @andrerom makes sense?

Reply:
File representing cache tag where you append cache keys?

Could do that, but then:
- more work (with race) to clean up relationship when tags are removed from an item _(unless we continue to view that as eventually consistent)
- can it hit file limits if it only appends and never cares if cache key is already there?
- _on invalidation probably need to rename like with redis to be able to iterate and clear without race possibilities_

So can work, but might present its own set of challenges.
Would it be better than replacing symlinks with just simple files that contain the cache key as value?

Reply:
(continued in https://github.com/symfony/symfony/issues/33924#issuecomment-539965850)